pkg/nsx/services/ipaddressallocation/ipaddressallocation.go

```diff
@@ -28,23 +28,25 @@ type IPAddressAllocationService struct {
 	VPCService               common.VPCServiceProvider
 }
 
-func InitializeIPAddressAllocation(service common.Service, vpcService common.VPCServiceProvider) (*IPAddressAllocationService, error) {
+func InitializeIPAddressAllocation(service common.Service, vpcService common.VPCServiceProvider, includeNCP bool) (*IPAddressAllocationService,
+	error) {
 	wg := sync.WaitGroup{}
 	wgDone := make(chan bool)
 	fatalErrors := make(chan error)
 
-	wg.Add(1)
-
 	ipAddressAllocationService := &IPAddressAllocationService{Service: service, VPCService: vpcService}
 	ipAddressAllocationService.ipAddressAllocationStore = &IPAddressAllocationStore{ResourceStore: common.ResourceStore{
 		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPAddressAllocationCRUID: indexFunc}),
 		BindingType: model.VpcIpAddressAllocationBindingType(),
 	}}
 
-	tags := []model.Tag{
-		{Scope: String(common.TagScopeIPAddressAllocationCRUID)},
+	if includeNCP {
+		wg.Add(2)
+		go ipAddressAllocationService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeIPAddressAllocation, []model.Tag{{Scope: String(common.TagScopeNCPCluster)}}, ipAddressAllocationService.ipAddressAllocationStore)
```

Comment:
It looks that you only check the tag.scope, not check the tag.tag. That will cause an issue that the store will include the other clusters' resources, right? In another word, in your test, your tag values for `ncp/cluster` and `nsx-op/cluster` are different (`d215b1fa-5432-498f-9214-d7da8bd4731d` vs `d215b1fa-5432-498f-9214-d7da8bd4731f`), the cleanup still picked all of them, I think it's not expected, it should only choose one specific cluster's resources.

Reply:
ok